Apps/Examples/Examples/All Examples/Custom2DPuckExample.swift

```diff
@@ -50,37 +142,72 @@ public class Custom2DPuckExample: UIViewController, ExampleProtocol {
+        button.backgroundColor = #colorLiteral(red: 0, green: 0.4784313725, blue: 0.9882352941, alpha: 1)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.layer.cornerRadius = 20
+        button.addTarget(self, action: #selector(changePuckOptions(sender:)), for: .touchUpInside)
+        view.addSubview(button)
+
+        // Set button location
+        let horizontalConstraint = button.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -24)
+        let verticalConstraint = button.centerXAnchor.constraint(equalTo: view.centerXAnchor)
+        let widthConstraint = button.widthAnchor.constraint(equalToConstant: 200)
+        let heightConstraint = button.heightAnchor.constraint(equalToConstant: 40)
+        NSLayoutConstraint.activate([horizontalConstraint, verticalConstraint, widthConstraint, heightConstraint])
     }
 
-    func syncPuckAndButton() {
-        // Update puck config
-        var configuration = Puck2DConfiguration(topImage: UIImage(named: "star"))
-        configuration.showsAccuracyRing = showsAccuracyRing
-        configuration.accuracyRingColor = UIColor.skyBlue
-        configuration.accuracyRingBorderColor = UIColor.lightGray
+    @objc public func changePuckOptions(sender: UIButton) {
+        let alert = UIAlertController(title: "Toggle Puck Options",
+                                      message: "Select an options to toggle.",
+                                      preferredStyle: .actionSheet)
+
+        alert.addAction(UIAlertAction(title: "Toggle Puck visibility", style: .default, handler: { [weak self] _ in
+            self?.showsPuck.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Toggle Puck image", style: .default, handler: { [weak self] _ in
+            self?.puckImage.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Toggle bearing visibility", style: .default, handler: { [weak self] _ in
+            self?.showsBearing.toggle()
+        }))
 
-        mapView.location.options.puckType = .puck2D(configuration)
+        alert.addAction(UIAlertAction(title: "Toggle accuracy ring", style: .default, handler: { [weak self] _ in
+            self?.showsAccuracyRing.toggle()
+        }))
 
-        // Update button title
-        let title: String = showsAccuracyRing ? "Disable Accuracy Radius" : "Enable Accuracy Radius"
-        toggleAccuracyRadiusButton.setTitle(title, for: .normal)
+        alert.addAction(UIAlertAction(title: "Toggle bearing source", style: .default, handler: { [weak self] _ in
+            self?.bearingSource.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Toggle Map Style", style: .default, handler: { [weak self] _ in
+            self?.style.toggle()
+        }))
+
+        alert.addAction(UIAlertAction(title: "Toggle Projection", style: .default, handler: { [weak self] _ in
+            self?.projection.toggle()
```

Comment:
we don't need a weak reference of `self` inside alert action's handler, given that we don't retain the alert controller.